processing/src/main/java/org/apache/druid/query/filter/DruidObjectPredicate.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.filter;
+
+import javax.annotation.Nullable;
+
+public interface DruidObjectPredicate<T>
+{
+  static <T> DruidObjectPredicate<T> alwaysFalse()
```

Comment:
This isn't really the right name, since it's not always false. So, we should change it to make the call sites more legible. Maybe a better name is `falseWithUnknownForNull()`?

Reply:
I guess I was trying to match the naming/behavior of `DruidLongPredicate.ALWAYS_FALSE` and other numeric predicates which also return unknown for `applyNull`, should I rename them all?

Reply:
I would, yes. It's not good for a thing named "always false" to not always return false, since people will get confused and use it where it isn't appropriate.